### PR TITLE
[WIP]Docker

### PR DESCRIPTION
**Pull request by m1guelpf**
This Pull Request should add a Docker version of OrgManager.

Reference for the docs (once it's merged):

``` markdown
docker run -d -p 8000:8000 orgmanager/orgmanager
```

ping @mastilver for revision

### CODE REVIEW

**Comment by mastilver** [comment]
What is your version of docker? (`docker -v`)

This is an API which is in rc, so you need to uninstall `docker` and run `curl -fsSL https://test.docker.com/ | sh`

**Comment by m1guelpf** [comment]
@mastilver `Docker version 17.04.0-ce, build 4845c56`.
How could I automate copying .env.example to .env, seting the .env variables with docker, composer install, yarn install, yarn run production & php artisan app:install ?

**Comment by mastilver** [comment]
I've got: `17.05.0-ce-rc2`, you need at least `17.05.0`

use `RUN` to run `yarn install` & co

check that for `ENV` https://docs.docker.com/engine/reference/builder/#env

**Comment by m1guelpf** [comment]
@mastilver And what about the MySQL database? What are the credentials?

**Comment by mastilver** [comment]
normally, you would need to use another image for the database
when running a container, you can use `--env` to set env variables

**Comment by m1guelpf** [comment]
@mastilver So we don't have a database?

**Comment by mastilver** [comment]
Not inside that container,
That container will only run your app, you can run another `mysql` container to connect to it

**Comment by m1guelpf** [comment]
@mastilver So then, maybe something like https://github.com/kyleferguson/laravel-with-docker-example would be better?

**Comment by mastilver** [comment]
If you look into that project, there is a `docker-compose.yml`, this is doing the link between your docker image and the mysql, redis, ... containers

**Comment by m1guelpf** [comment]
@mastilver So the fix would be to add a docker compose file to this PR?

**Comment by mastilver** [comment]
Yeah, I should have mention it earlier but yes!

Check those to learn more about docker:
- https://github.com/veggiemonk/awesome-docker#where-to-start
- https://github.com/veggiemonk/awesome-docker#docker-compose-file

**Comment by m1guelpf** [comment]
@mastilver So, https://github.com/orgmanager/orgmanager/pull/255/commits/dd3f57c1690184c88c90053f6668a58aaf0ab6ce would add a database and auto-configure the database credentials both for the application and for the server?

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      - DB_HOST=mysql
+      - DB_DATABASE=orgmanager
+      - DB_USERNAME=root
+      - DB_PASSWORD=root
+    depends_on:
+      - mysql
+      - redis
+      - mail
+    volumes:
+      - ".:/var/www/html"
```
Comment (mastilver) [inline docker-compose.yml]: not sure about that...?
Why is that needed?

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    volumes:
+      - ".:/var/www/html:ro"
+      - "./nginx:/etc/nginx/conf.d:ro"
+    depends_on:
+      - php
+    ports:
+      - "8080:80"
+
+  php:
+    image: greut/laravel:latest
```
Comment (mastilver) [inline docker-compose.yml]: why not using your image?

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
+version: '2'
+
+services:
+  nginx:
+    image: nginx:1.11-alpine
+    volumes:
+      - ".:/var/www/html:ro"
+      - "./nginx:/etc/nginx/conf.d:ro"
```
Comment (mastilver) [inline docker-compose.yml]: needs to be updated!

Comment (m1guelpf) [inline docker-compose.yml]: What needs to be updated?

File: docker-compose.yml (hunk 2 above)
Comment (m1guelpf) [inline docker-compose.yml]: You say, specify there a relative path to the Dockerfile or something like orgmanager/orgmanager when I upload it to the Docker hub?

File: docker-compose.yml (hunk 1 above)
Comment (m1guelpf) [inline docker-compose.yml]: To mount the current dir to the nginx folder?

File: docker-compose.yml (hunk 3 above)
Comment (mastilver) [inline docker-compose.yml]: Sorry was on my phone,
Here you are trying to copy the nginx folder (which doesn't exist...) into the file /etc/nginx/conf.d

File: docker-compose.yml (hunk 2 above)
Comment (mastilver) [inline docker-compose.yml]: I would say relative path, up to you

File: docker-compose.yml (hunk 3 above)
Comment (m1guelpf) [inline docker-compose.yml]: Can I download the conf.d file from a gist instead of localy?

Content:
```
server {
    index index.php index.html;

    root /var/www/html/public;

    location / {
        try_files $uri $uri/ /index.php?q=$uri&$args;
    }

    location ~ \.php$ {
        try_files $uri =404;
        fastcgi_split_path_info ^(.+\.php)(/.+)$;
        fastcgi_pass php:9000;
        fastcgi_index index.php;
        include fastcgi_params;
        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
        fastcgi_param PATH_INFO $fastcgi_path_info;
    }
}
```

File: docker-compose.yml (hunk 2 above)
Comment (m1guelpf) [inline docker-compose.yml]: So I would specify `.` or `Dockerfile` or `./Dockerfile`?

File: docker-compose.yml (hunk 3 above)
Comment (mastilver) [inline docker-compose.yml]: Why not putting it into your repo...?

Comment (mastilver) [inline docker-compose.yml]: https://github.com/llitfkitfk/docker-compose-demo/tree/master/nginx-proxy

File: docker-compose.yml (hunk 2 above)
Comment (mastilver) [inline docker-compose.yml]: https://github.com/llitfkitfk/docker-compose-demo/blob/25d0316fb338f4cebf3b0c8e101ccad0e7537543/symfony/docker-compose.yml#L27

Comment (m1guelpf) [inline docker-compose.yml]: So I'd just put `.`, as the Dockerfile is in the root?

File: docker-compose.yml (hunk 3 above)
Comment (m1guelpf) [inline docker-compose.yml]: Maybe that line can be deleted and use the default nginx config...

**Comment by m1guelpf** [comment]
@mastilver Is it OK now?

**Comment by mastilver** [comment]
Looks alright, Have you actually tried it???

**Comment by m1guelpf** [comment]
@mastilver Updating to Ubuntu 117.04... will try it when it finishes
